api/src/User/Repository/UserRepository.ts

```diff
@@ -0,0 +1,43 @@
+import { EntityRepository, Repository } from 'typeorm'
+import UserEntity from '../Entity/UserEntity'
+import RegisterDto from '../../Authentification/DTO/RegisterDto'
+import PasswordHelper from '../../Authentification/helpers/PasswordHelper'
+import { ConflictException, InternalServerErrorException } from '@nestjs/common'
+import { JwtDTO } from '../../Authentification/DTO/JwtDTO'
+import UserInfoEntity from '../Entity/UserInfoEntity'
+import * as bcrypt from 'bcrypt'
+
+@EntityRepository(UserEntity)
+export default class UserRepository extends Repository<UserEntity> {
+  async register({ email, password }: RegisterDto): Promise<boolean> {
+    const user = new UserEntity()
+    user.email = email
+    user.salt = await bcrypt.genSalt()
+    user.password = await PasswordHelper.hash(password, user.salt)
+    try {
+      const userInfo = new UserInfoEntity()
+      await userInfo.save()
+      user.userInfo = userInfo
+      await user.save()
+    } catch (e) {
+      //error code for already exist
+      if (e.errno === 19 || e.code === '23505')
```

Comment:
Handler with switch case perhaps if we have a lot of e.code handling different but we can check this later

Reply:
switch case wont work for multi e possibility, since they'll have different code depending on the db technology